internal/pkg/monitor/monitor.go

```diff
@@ -213,10 +262,20 @@ func (m *monitorT) notify(ctx context.Context, hits []es.HitT) {
 		maxVal := hits[sz-1].SeqNo
 		m.storeCheckpoint(maxVal)
 
-		select {
-		case m.outCh <- hits:
-		case <-ctx.Done():
+		m.mut.RLock()
+		var wg sync.WaitGroup
+		wg.Add(len(m.subs))
+		for _, s := range m.subs {
+			go func(s *subT) {
```

Comment:
depending on the usage, there is a chance of the race and go routine leak:
1. you start a go routine in the loop to send the message
2. the client unsubscribes, never consumes the message from the channel
3. by the time the go routine fires it will be blocked on sending the hit and leak. the unbuffered channel can make this happen more often 

with this addition the monitor became heavier since we added additional syncing, kicking off the go routines for each notifications and could potentially leak go routines.
could help a bit if using buffered channel, and maybe drop notifications if channel is full to avoid the go routine leak.

Reply:
same problem with the Mock

Reply:
The solution is to close the channel on unsubscribe. Then the sending goroutine will panic for sending on a closed goroutine, the panic can then be recovered and the goroutine will exit. On the next notify the subscription will be gone and that issue will not be hit.

I have added a tests to confirm this behavior.

Reply:
Still not a big fan of the locking and go routines and now panic recovery, but ok.